### PR TITLE
INSPECT-11552 Update jsonwebtoken

### PR DESCRIPTION
**Pull request by xragon**
## Summary

- update to latest possible version of `jsonwebtoken`
- Set `allowInsecureKeySizes` as per https://github.com/auth0/node-jsonwebtoken/wiki/Migration-Notes:-v8-to-v9 in order to maintain current behavior
- Updated the pipeline to check tests
- Updated devDependencies to ensure node 22 functionality
- Updated major version to avoid any accidental upgrades

## Problem description

- Security alerts are being generated in services that consume this package due to the old version of `jsonwebtoken`

See https://github.com/SafetyCulture/base-rest-client/pull/55

## Checklist

- [x] Is this PR a reasonable size?

- [ ] List deployment sequence with all relevant PRs.
  - ...

- [ ] For breaking changes: I have confirmed that other repos are not affected, or I have planned a deployment sequence to address any impact.

---

**Code Review Guidelines** for Reviewers

- Try to review in a timely manner. Opinions/nitpicks should not be blockers. Pair on a call for non-trivial feedback.
- Overall design and approach should follow established patterns. Don't try to make the PR perfect.
- Try to identify edge cases, race conditions, over-engineering, lack of test coverage and complexity.
- If you don't feel qualified to review the code, pass it on to someone who is.

### CODE REVIEW

File: lib/jwt-authentication/json-web-token.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ module.exports = {
         var options = {
             // these are checked by jwt-claims-validator
             ignoreExpiration: true,
-            ignoreNotBefore: true
+            ignoreNotBefore: true,
+            clockTolerance: 30 // Allow 30 seconds of leeway for clock skew
```
Comment (sc-peter) [inline lib/jwt-authentication/json-web-token.js]: I suggest setting this to 60 seconds as that's what we're using elsewhere https://github.com/SafetyCulture/api-identity/blob/192f251f16958dff11713f7319fe9bbc4723b1cc/internal/security/jwt/verifier/verifier.go#L21

Update: that is for S12JWT, I can't find what we set as a leeway for ASAP. Feel free to keep it at 30 seconds.

File: lib/jwt-authentication/json-web-token.js [hunk 2]
```diff
@@ -39,9 +40,8 @@ module.exports = {
         var jsonWebTokenOptions = {
             algorithm: 'RS256',
             expiresIn: options.expiresInSeconds || 30,
-            headers: {
-                kid: options.kid
-            }
+            keyid: options.kid,
+            allowInsecureKeySizes: true // Allow small key sizes for test environments
```
Comment (sc-peter) [inline lib/jwt-authentication/json-web-token.js:44]: Is there a way to only allow this in test environments?

Comment (sc-peter) [reply lib/jwt-authentication/json-web-token.js:44]: Maybe we can do this based on an env var? You can check for `SC_ENVIRONMENT = development` and only set  it to true in that case potentially? That var is set in dev and when running it locally via intercept.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM 288761735648.dkr.ecr.us-east-1.amazonaws.com/safetyculture/node:22-builder-alpine-3.19 AS builder
```
Comment (bhargysh-sc) [inline Dockerfile:1]: Could we add run instructions for this in the README?

Comment (xragon) [reply Dockerfile:1]: Happy to update the readme, but not sure what you are asking for here? This is for the PR checks/build pipeline, it will do an install and run the tests to ensure everything is okay automatically. There's nothing you need to do locally.

image

**Comment by xragon** [comment]
@bhargysh-sc @sc-peter after reading peters comments, I have put together this PR https://github.com/SafetyCulture/jwt-authentication/pull/16 where the `allowInsecureKeySizes` flag has been removed. This does mean a possible behavioral change, so I am working on getting this setup to be able to test with a service in sandpit.

File: Dockerfile (hunk 3 above)
Comment (bhargysh-sc) [reply Dockerfile:1]: Ah sorry I assumed it was for local running, all good

**Comment by xragon** [comment]
See https://github.com/SafetyCulture/jwt-authentication/pull/16 instead
